### ISSUE TITLE
Prepending links and images with a base folder

### ISSUE THREAD

**Issue by DannyBen** [davidfstr/rdiscount#141]
I tried [looking for an answer on StackOverflow][1], but I guess this feature does not exist, so I am posting here more as a feature request (or perhaps it is already possible?).

My markdown documents contain links (or images) that are relative to the path of the markdown file itself.

It would be nice if there was a way to tell RDiscount that it should prepend all relative links with a string (folder) of my choice?

I am trying to achieve an effect similar to how GitHub shows images in README - where the files are looked for in the same directory as the README.

Here is an example code:

```ruby
require 'rdiscount'

markdown = "![pic](image.png)\n\n[link](somewhere.html)"

doc = RDiscount.new(markdown)
# I would like to do something here, like:
# doc.base_link_path = 'SOMEFOLDER'
html = doc.to_html
puts html

# actual output => 
#  <p><img src="image.png" alt="pic" /></p>
#  <p><a href="somewhere.html">link</a></p>"

# desired output => 
#  <p><img src="SOMEFOLDER/image.png" alt="pic" /></p>
#  <p><a href="SOMEFOLDER/somewhere.html">link</a></p>"
```

[1]: https://stackoverflow.com/questions/50281044/injecting-base-directory-to-rdiscount-markdown-conversion

**Comment by DannyBen** [comment]
Yes - I was trying to do something different, but ended up opting for simplicity.

Had two use cases that needed this:

1. Wanted to allow same flexibility as GitHub, meaning to also allow using absolute paths in places where the root path of the markdown document is not necessarily the root path of the webserver.
2. I have a Sinatra markdown webserver ([madness](https://github.com/dannyben/madness)) that converts the file `somefolder/README.md` to this web address: `/somefolder`. The only way for images to load in this case, is if I use a trailing slash: `/somefolder/` - which was less desireable.

All the experiments can be seen [in this repo](https://github.com/DannyBen/temp)

The bottom line is that I was looking (and still think it has usefulness) for a way to intercept links (or other tags) and update the result.  CommonMarker [has a `doc.walk` method](https://github.com/gjtorikian/commonmarker#example-walking-the-ast).

**Comment by davidfstr** [comment]
RDiscount is a fairly thin wrapper around the underlying [Discount](http://www.pell.portland.or.us/~orc/Code/discount/) library.

I am not aware of a Discount flag or setting that can change the assumed base URL for generating links. Therefore to get that functionality you'd have to advocate to the Discount project for that capability. Once it is in Discount, I can look at exposing the functionality in RDiscount.

**Comment by DannyBen** [comment]
Thanks. You can close this issue as far as I am concerned. I have resolved it by first switching to CommonMarker for its `walk` function, but eventually, simply removing the need to alter links in the first place.
